### ISSUE TITLE
Can not find pyconfig.h in ubuntu 14.04 use NDK r10c(devel branch)

### ISSUE THREAD

**Issue by yuanguangyong** [moritz-wundke/Boost-for-Android#97]
Dear Moritz-wundke and all friends:
I try to build boost for android in ubuntu 14.04.
My NDK version is r10c, so I use the devel branch.
1、Step1: First time, I get the error:
gcc.compile.c++ bin.v2/libs/python/build/gcc-androidR8e/release/link-static/threading-multi/numeric.o
In file included from ./boost/python/detail/prefix.hpp:13:0,
                 from ./boost/python/numeric.hpp:8,
                 from libs/python/src/numeric.cpp:6:
./boost/python/detail/wrap_python.hpp:50:23: fatal error: pyconfig.h: No such file or directory
fatal error: pyconfig.h: No such file or directory,there have 28 same errors.
I get the suggest:
sudo apt get install phthon-dev
After I install the phthon, I get the new error:
2、Step2:
gcc.compile.c++ bin.v2/libs/python/build/gcc-androidR8e/release/link-static/threading-multi/numeric.o
In file included from ./boost/python/detail/wrap_python.hpp:50:0,
                 from ./boost/python/detail/prefix.hpp:13,
                 from ./boost/python/numeric.hpp:8,
                 from libs/python/src/numeric.cpp:6:
/usr/include/python2.7/pyconfig.h:15:52: fatal error: arm-linux-gnueabi/python2.7/pyconfig.h: No such file or directory compilation terminated.

 Fatal error: arm-linux-gnueabi/python2.7/pyconfig.h,there is 28 same error.

In /usr/include/python2.7/pyconfig.h: Line15:
# elif defined(__ARM_EABI__) && !defined(__ARM_PCS_VFP)
#  include <arm-linux-gnueabi/python2.7/pyconfig.h>

I have not got good suggest in the internet.
Can you help me ,  what's wrong with me ?
Thanks.

**Comment by Bendr0id** [comment]
Hi @ll,

i've exactly the same problem.

gcc.compile.c++ bin.v2/libs/python/build/gcc-androidR8e/release/link-static/threading-multi/numeric.o
In file included from ./boost/python/detail/wrap_python.hpp:50:0,
                 from ./boost/python/detail/prefix.hpp:13,
                 from ./boost/python/numeric.hpp:8,
                 from libs/python/src/numeric.cpp:6:
/usr/include/python2.7/pyconfig.h:15:52: fatal error: arm-linux-gnueabi/python2.7/pyconfig.h: No such file or directory
compilation terminated.

Enviroment is Ubuntu 14.10 and android-ndk-r9c

Is there any solution for this?

**Comment by moritz-wundke** [comment]
Have you tried it in a different OS?

On Wed, Feb 4, 2015 at 11:20 AM, Ben Gräf <notifications@github.com> wrote:

> Hi @ll <https://github.com/ll>,
>
> i've exactly the same problem.
>
> gcc.compile.c++
> bin.v2/libs/python/build/gcc-androidR8e/release/link-static/threading-multi/numeric.o
> In file included from ./boost/python/detail/wrap_python.hpp:50:0,
> from ./boost/python/detail/prefix.hpp:13,
> from ./boost/python/numeric.hpp:8,
> from libs/python/src/numeric.cpp:6:
> /usr/include/python2.7/pyconfig.h:15:52: fatal error:
> arm-linux-gnueabi/python2.7/pyconfig.h: No such file or directory
> compilation terminated.
>
> Enviroment is Ubuntu 14.10 and android-ndk-r9c
>
> Is there any solution for this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MysticTreeGames/Boost-for-Android/issues/97#issuecomment-72831257>
> .
>



-- 
*Moritz Wundke*
www.MoritzWundke.com

**Comment by Bendr0id** [comment]
Tested on debian as well.

But adding "-without-python \" to  ./bjam call in build-android.sh fixed it for me.

**Comment by moritz-wundke** [comment]
If we can confirm the fix and no side effects will appear I'm OK with
integrating it.

On Wed, Feb 4, 2015 at 1:30 PM, Ben Gräf <notifications@github.com> wrote:

> Tested on debian as well.
>
> But adding "-without-python \" to ./bjam call in build-android.sh fixed it
> for me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MysticTreeGames/Boost-for-Android/issues/97#issuecomment-72846669>
> .
>



-- 
*Moritz Wundke*
www.MoritzWundke.com

**Comment by Aman5692** [comment]
Hi Bendr0id,
i'm facing the same problem while building boost1.55 with androidndkr10d on ubuntu 14.04.
I added -without-python \ on line no 142 of build-android.sh ...... but it does not seem to solve the error, can u tell me the fix.
Thanks

**Comment by Bendr0id** [comment]
Hi Aman,

check line 412.

There is something like this:
```
  { ./bjam -q                         \
         target-os=linux              \
         toolset=$TOOLSET             \
         $cxxflags                    \
         link=static                  \
         threading=multi              \
         --layout=versioned           \
         -sICONV_PATH=`pwd`/../libiconv-libicu-android/armeabi \
         -sICU_PATH=`pwd`/../libiconv-libicu-android/armeabi \
         --prefix="./../$BUILD_DIR/"  \
         $LIBRARIES                   \
         install 2>&1                 \
         || { dump "ERROR: Failed to build boost for android!" ; exit 1 ; }
  } | tee -a $PROGDIR/build.log
```


change it to:
```
  { ./bjam -q                         \
         target-os=linux              \
         toolset=$TOOLSET             \
         $cxxflags                    \
         link=static                  \
         threading=multi              \
         --layout=versioned           \
         -without-python \
         -sICONV_PATH=`pwd`/../libiconv-libicu-android/armeabi \
         -sICU_PATH=`pwd`/../libiconv-libicu-android/armeabi \
         --prefix="./../$BUILD_DIR/"  \
         $LIBRARIES                   \
         install 2>&1                 \
         || { dump "ERROR: Failed to build boost for android!" ; exit 1 ; }
  } | tee -a $PROGDIR/build.log
```

**Comment by Aman5692** [comment]
Well i did the same, but it terminates with this

Invalid option: -w

usage: ./bjam [ options ] targets...

-a      Build all targets, even if they are current.
-dx     Set the debug level to x (0-9).
-fx     Read x instead of Jambase.
-jx     Run up to x shell commands concurrently.
-lx     Limit actions to x number of seconds after which they are stopped.
-mx     Maximum target output saved (kb), default is to save all output.
-n      Don't actually execute the updating actions.
-ox     Write the updating actions to file x.
-px     x=0, pipes action stdout and stderr merged into action output.
-q      Quit quickly as soon as a target fails.
-sx=y   Set variable x=y, overriding environment.
-tx     Rebuild x, even if it is up-to-date.
-v      Print the version of jam and exit.
--x     Option is ignored.

ERROR: Failed to build boost for android!
Done!

**Comment by Bendr0id** [comment]
hmm could be a missing "-"

please try:

`--without-python`

**Comment by Aman5692** [comment]
Thanks Bendr0id,
that extra "-" did the trick

**Comment by Bendr0id** [comment]
;-)

**Comment by portwaypoint** [comment]
Same problem here; Can confirm fix works (Boost 1.55 on NDK9 within Android AOSP KitKat 4.4).
